### PR TITLE
trace counter

### PR DESCRIPTION
**Pull request by cwbhhjl**

### CODE REVIEW

**Review by yann-sjtu — APPROVED** [review]

File: src/tracer/mod.rs [hunk 1]
```diff
@@ -111,6 +123,18 @@ impl Tracer {
             .unwrap()
             .clone()
     }
+
+    pub fn count(&mut self) {
```
Comment (xgaozoyoe) [inline src/tracer/mod.rs:127]: maybe inc_trace_size()? since inc means add one. count does not imply the number.

File: src/tracer/mod.rs [hunk 2]
```diff
@@ -54,13 +54,23 @@ pub struct Tracer {
     // Wasm Image Function Idx
     pub wasm_input_func_idx: Option<u32>,
     pub wasm_input_func_ref: Option<FuncRef>,
+    only_counter: bool,
+    counter: usize,
```
Comment (xgaozoyoe) [inline src/tracer/mod.rs:58]: counter = trance_size might be better?

File: src/tracer/phantom.rs [hunk 3]
```diff
@@ -56,14 +56,27 @@ impl Tracer {
         wasm_input_function_idx: u32,
         keep_value: Option<u64>,
     ) {
+        let has_return_value = callee_sig.return_type().is_some();
+
+        if self.only_count() {
```
Comment (xgaozoyoe) [inline src/tracer/phantom.rs:61]: can we make count_only as an argument of fill_trace?

File: src/runner.rs [hunk 4]
```diff
@@ -238,13 +238,26 @@ impl Interpreter {
...
+                self.tracer.clone()
+            } else {
+                None
+            }
         } else {
             None
         }
     }
 
+    fn tracer_count(&self) {
```
Comment (xgaozoyoe) [inline src/runner.rs:252]: inc_tracer_if_not_masked

File: src/runner.rs [hunk 5]
```diff
@@ -238,13 +238,26 @@ impl Interpreter {
     }
 
     fn get_tracer_if_active(&self) -> Option<Rc<RefCell<Tracer>>> {
```
Comment (xgaozoyoe) [inline src/runner.rs:240]: name of the function does not imply the logic clearly.  the logic is get tracer if neither masked or in counting phase?

File: src/tracer/mod.rs [hunk 6]
```diff
@@ -54,13 +54,23 @@ pub struct Tracer {
     // Wasm Image Function Idx
     pub wasm_input_func_idx: Option<u32>,
     pub wasm_input_func_ref: Option<FuncRef>,
+    only_counter: bool,
```
Comment (xgaozoyoe) [inline src/tracer/mod.rs:57]: phase: TracePhase and let TracePhase = Counting (usize) | Normal

File: src/tracer/mod.rs (hunk 2 above)
Comment (xgaozoyoe) [reply src/tracer/mod.rs:58]: or put counter in TracePhase? Better put only_counter and counter in one field

**Review by junyu0312 — APPROVED** [review]

**Pull request merged**